web/site/js/vis/LogWindow.js

```diff
@@ -0,0 +1,95 @@
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import * as PIXI from "pixi.js-legacy";
+import VObject from "./VObject";
+import {Scrollbox} from 'pixi-scrollbox'
+
+
+const LOG_TEXT_STYLE = {
+    fill: "#0052ff",
+    fontFamily: "Verdana",
+    fontSize: 11,
+    fontStyle: "italic",
+    fontWeight: "normal"
+};
+
+const LOG_TEXT_LINE_HEIGHT = 15;
+const LOG_WINDOW_MAX_SIZE = 500;
+export const LOG_WINDOW_WIDTH = 400;
+
+export default class LogWindow extends VObject {
+    constructor() {
+        super();
+
+        let height = window.innerHeight - 100;
+        this.logIndex = 0;
+
+        this._root = new Scrollbox({
+            boxWidth: LOG_WINDOW_WIDTH,
+            boxHeight: height,
+            fade: true,
+            overflowX: "auto",
+            overflowY: "auto"
+        });
+        this.logContainer = new PIXI.Container()
+        this.lastline = new PIXI.Graphics()
+        this.lastline.clear();
+        this.lastline.beginFill(0xFFFFFF);
+        // this.lastline.lineStyle(0);
+        this.lastline.drawRect(0, 0, LOG_WINDOW_WIDTH, LOG_TEXT_LINE_HEIGHT);
+        this.lastline.endFill();
+        this.logContainer.addChild(this.lastline)
+
+        this._root.content.addChild(this.logContainer)
+        this._root.update();
+        this.loglist = [];
+    }
+
+    addLog(text, color = "#0052ff") {
+        if (this.loglist.length === LOG_WINDOW_MAX_SIZE) {
+            let rm = this.loglist.shift();
+            this.logContainer.removeChild(rm)
+        }
+
+        LOG_TEXT_STYLE.fill = color;
+        let log = new PIXI.Text(text, LOG_TEXT_STYLE);
+        log.position.set(3, 3 + this.logIndex * LOG_TEXT_LINE_HEIGHT);
+        this.logIndex++;
+        this.logContainer.addChild(log);
+        this.loglist.push(log);
+
+        this.logContainer.position.set(0, -this.loglist[0].position.y);
+        this.lastline.position.set(0, log.y + LOG_TEXT_LINE_HEIGHT)
+
+        this._root.resize({boxWidth: LOG_WINDOW_WIDTH, boxHeight: this._root.boxHeight});
+        this._root.ensureVisible(0, log.y + this.logContainer.y, LOG_WINDOW_WIDTH, log.height);
+    }
+
+    resetLayout(width, height) {
+        this._root.resize({boxWidth: LOG_WINDOW_WIDTH, boxHeight: height - 100})
```

Comment:
Put this `100` and the same number in line 48 in a constant? E.g. `LOG_WINDOW_VERTICAL_EDGE_PADDING`

Reply:
added constant